lib/extensions.js

```diff
@@ -260,6 +261,80 @@ function getBinaryPath() {
 }
 
 /**
+ * An array of paths suitable for use as a local disk cache of the binding.
+ *
+ * @return {[]String} an array of paths
+ * @api public
+ */
+function getCachePathCandidates() {
+  return [
+    process.env.npm_config_sass_binary_cache,
+    process.env.npm_config_cache,
+    getBinaryPath(),
+  ].filter(function(_) { return _; });
+}
+
+/**
+ * The most suitable location for caching the binding on disk.
+ *
+ * Given the candidates directories provided by `getCachePathCandidates()` this
+ * returns the first writable directory. By treating the candidate directories
+ * as a prioritised list this method is deterministic, assuming no change to the
+ * local environment.
+ *
+ * @return {String} directory to cache binding
+ * @api public
+ */
+function getBinaryCachePath() {
+  var i,
+    cachePath,
+    cachePathCandidates = getCachePathCandidates();
+
+  for (i = 0; i < cachePathCandidates.length; i++) {
```

Comment:
Expanding on this, `forEach` is intended to creating side effects. This is closer to a `reduce` but that also really messy to exit early from.